v5/core/base_service.go

```diff
@@ -581,22 +608,63 @@ func getErrorMessage(responseMap map[string]interface{}, statusCode int) string
 // configuration, and then set it on the service instance.
 // If maxRetries and/or maxRetryInterval are specified as 0, then default values
 // are used instead.
+//
```

Comment:
Thought it might be good to add these details since it took me a while to work through the various datastructures
and figure this out :)